### ISSUE TITLE
Usage of findDOMNode

### ISSUE THREAD

**Issue by Methuselah96** [DevExpress/devextreme-reactive#2727]
- [x] I have searched this repository's [issues](https://github.com/devexpress/devextreme-reactive/issues) and believe that this is not a duplicate.
  - I found https://github.com/DevExpress/devextreme-reactive/issues/1602#issuecomment-445241666, but it seems unresolved.

### I'm using ...

- [x] React Grid
- [ ] React Chart
- [ ] React Scheduler

### Current Behaviour

I'm using a `VirtualTable` within a `Grid` from `@devexpress/dx-react-grid-material-ui` and when I run React in StrictMode it produces this warning:

```
react-dom.development.js?4a22:88 Warning: findDOMNode is deprecated in StrictMode. findDOMNode was passed an instance of RefHolder which is inside StrictMode. Instead, add a ref directly to the element you want to reference. Learn more about using refs safely here: https://fb.me/react-strict-mode-find-node
    in div (created by TableContainerBase)
    in TableContainerBase (created by WithStyles(TableContainerBase))
    in WithStyles(TableContainerBase) (created by Sizer)
    in RefHolder (created by Sizer)
    in Sizer (created by GridSizer)
    in GridSizer (created by VirtualTableLayout)
    in VirtualTableLayout (created by TableLayoutBase)
    in TableLayoutBase (created by VirtualTableLayout)
    in VirtualTableLayout (created by RenderComponent)
    in RenderComponent (created by TemplatePlaceholderBase)
    in TemplatePlaceholderBase (created by Context.Consumer)
    in Unknown (created by Context.Consumer)
    in Unknown (created by TemplateConnectorBase)
    in TemplateConnectorBase (created by TemplatePlaceholderBase)
    in TemplatePlaceholderBase (created by Context.Consumer)
    in Unknown (created by Context.Consumer)
    in Unknown (created by TemplateConnectorBase)
    in TemplateConnectorBase (created by TableBase)
    in TemplatePlaceholderBase (created by Context.Consumer)
    in Unknown (created by Context.Consumer)
    in Unknown (created by TableBase)
    in TemplatePlaceholderBase (created by Context.Consumer)
    in Unknown (created by Context.Consumer)
    in Unknown (created by GridCore)
    in div (created by RootBase)
    in RootBase (created by WithStyles(RootBase))
    in WithStyles(RootBase) (created by GridCore)
    in TemplatePlaceholderBase (created by Context.Consumer)
    in Unknown (created by Context.Consumer)
    in Unknown (created by PluginHostBase)
    in PluginHostBase (created by GridBase)
    in GridBase (created by Grid)
    in Grid (created by NavigationPanel)
```

This error seems to originate from here: https://github.com/DevExpress/devextreme-reactive/blob/ce6b2612590b37b2be8a35ada8c4890f269fe8e6/packages/dx-react-core/src/sizer.tsx#L118

### Expected Behaviour

I would expect it not to produce an error.

### Steps to Reproduce

Link to the example that reproduces the issue: https://codesandbox.io/s/devextreme-react-grid-for-material-ui-yoj50?fontsize=14&hidenavigation=1&theme=dark (you can fork one of our pre-configured [examples on CodeSandbox](https://github.com/DevExpress/devextreme-reactive/issues/2358)).

### Screenshots

N/A

### Environment

- devextreme-reactive: 2.4.1
- react: 16.13.0
- browser: N/A
- bootstrap: none
- react-bootstrap: none
- material-ui: 4.9.4

**Comment by ushkal** [comment]
Hi @Methuselah96 ,

Thank you for reporting this problem. We'll fix it in the next major version.

**Comment by Methuselah96** [comment]
@ushkal Okay, thanks. Just for purposes of my own decision-making, is there any expected timeline for the next major version release?

**Comment by Methuselah96** [comment]
@ushkal Also out of curiosity, is there a reason why it needs to wait for a major version release?

**Comment by ushkal** [comment]
> is there any expected timeline for the next major version release? Back in v1, it was stated that it would be fixed in v2 ([#1602 (comment)](https://github.com/DevExpress/devextreme-reactive/issues/1602#issuecomment-445241666)), but that hasn't seemed to happen

Unfortunately, we cannot provide you with an exact timeframe. It depends on whether or not will the React team decides to remove the findDOMNode method from the library.

**Comment by Methuselah96** [comment]
@ushkal Would you accept a PR that removes `findDOMNode`? Usage of `findDOMNode` will be a blocker for adopting blocking mode or concurrent mode in the future (https://reactjs.org/docs/concurrent-mode-adoption.html#feature-comparison).

**Comment by ushkal** [comment]
@Methuselah96 Please accept my apologies for the delayed response. Yes, we'll review and accept such a pull request.

**Comment by JollyGoodHolly** [comment]
@Methuselah96 @ushkal Any update on this issue/PR or the release timeline for the next major version?

**Comment by Methuselah96** [comment]
No, I haven't had any time to work on it.

**Comment by MaximKudriavtsev** [comment]
Hi @Methuselah96, @JollyGoodHolly,

Unfortunately, we can't fix this behavior in the current version because this will cause a Breaking Change. We'll fix it in the next major version of our product.

**Comment by nextglabs** [comment]
Any news about this @MaximKudriavtsev ?

**Comment by MaximKudriavtsev** [comment]
Hi @nextglabs,

We are working on version 3 (the next major) of Devextreme Reactive and plan to publish it after React v17 is released.

**Comment by vtrphan** [comment]
well React 17 has been released. when will v3 be released?

**Comment by MaximKudriavtsev** [comment]
@vtrphan,

You are right! Now we are working on [migration to React v17](https://github.com/DevExpress/devextreme-reactive/pull/3148). We will publish v3 of DevExtreme Reactive as soon as we complete it.
